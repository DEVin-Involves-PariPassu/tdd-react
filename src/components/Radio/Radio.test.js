import 'jest-styled-components'
const { render, screen } = require('@testing-library/react')
const { default: Radio } = require('.')

describe('<Radio />', () => {
  it('should render the radio', () => {
    const { container } = render(<Radio label="Radio" labelFor="check" value="anyValue"/>)

    const label = screen.getByText('Radio');
    expect(label).toBeInTheDocument();
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot();
  })
})