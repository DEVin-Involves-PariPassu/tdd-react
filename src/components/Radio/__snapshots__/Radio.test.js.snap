// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> should render the radio 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 1.8rem;
  height: 1.8rem;
  border: 0.2rem solid #de0202;
  border-radius: 50%;
  background: transparent;
  -webkit-transition: background 0.1s ease-in-out;
  transition: background 0.1s ease-in-out;
  outline: none;
  cursor: pointer;
}

.c1:before {
  content: '';
  width: 0.8rem;
  height: 0.8rem;
  border-radius: 50%;
  background: #de0202;
  opacity: 0;
  -webkit-transition: opacity 0.1s ease-in-out;
  transition: opacity 0.1s ease-in-out;
  position: absolute;
}

.c1:checked:before {
  opacity: 1;
}

.c2 {
  padding-left: 0.8rem;
  color: black;
  line-height: 1;
}

<div
  class="c0"
>
  <input
    class="c1"
    id="check"
    type="radio"
    value="anyValue"
  />
  <label
    class="c2"
    for="check"
  >
    Radio
  </label>
</div>
`;
